components/SelectOptionDetailed.tsx

```diff
@@ -0,0 +1,65 @@
+export enum Flag {
+  Danger,
+  Warning,
+  OK,
+}
+
+type OptionText = {
+  text: string;
+  flag?: Flag;
+};
+
+type OptionDetails = {
+  [label: string]: OptionText;
+};
+
+const setFlagColorText = (flag?: Flag) => {
+  switch (flag) {
+    case Flag.Danger:
+      return 'text-red';
+    case Flag.Warning:
+      return 'text-orange';
+    case Flag.OK:
+      return 'text-green';
+    default:
+      return '';
+  }
+};
+
+const SelectOptionDetailed = ({
+  title,
+  details,
+  diffValue,
+}: {
+  title: string;
+  details: OptionDetails;
+  diffValue?: OptionText;
+}) => {
+  return (
+    <div className="flex flex-col">
+      <div className="mb-0.5">{title}</div>
+
+      <div className="flex flex-col">
+        {Object.entries(details).map(([label, option]) => (
+          <div
+            key={option.text}
+            className={`space-y-0.5 text-xs text-fgd-3 ${setFlagColorText(
+              option.flag,
+            )}`}
+          >
+            {`${label}: ${option.text}`}
+          </div>
+        ))}
+        {diffValue && (
```

Comment:
What the `diffValue` means please? I don't get it

Reply:
bad variable naming